zombies.js

```diff
@@ -79,135 +94,211 @@
+   */
+
+  this.discardItem = function(item) {
+    if (this._pack.indexOf(item) !== -1) {
+      this._pack.splice(this._pack.indexOf(item), 1);
+      console.log("Item was discarded");
+      return true;
+    }
+    return false;
+  };
+
+  /**
+   * Player Class Method => equip(itemToEquip)
+   * -----------------------------
+   * Player equips a weapon item.
+   *
+   * Player can only equip Weapon instances.
+   * Player can only equip weapon items from their pack.
+   *
+   * If the player already has a weapon equipped (the equipped property
+   *   is set to an Item), find the itemToEquip in the pack and replace
+   *   it with the currently equipped item.  Then set the equipped property
+   *   to the itemToEquip.
+   * However, if the player doesn't already have a weapon equipped, simply
+   *   equip that item and remove it from the pack.
+   * You should be able to invoke this function on a Player instance.
+   *
+   * @name equip
+   * @param {Weapon} itemToEquip  The weapon item to equip.
+   */
+
+  this.equip = function(itemToEquip) {
+    let isWeapon = itemToEquip instanceof Weapon;
+    let currentWeapon = this.equipped;
+    if (!isWeapon || this._pack.indexOf(itemToEquip) === -1) {
+      return false;
+    }
+    if (currentWeapon) {
+      this._pack.push(currentWeapon);
+    }
+    this._pack.splice(this._pack.indexOf(itemToEquip), 1);
+    this.equipped = itemToEquip;
+  };
+
+  /**
+   * Player Class Method => eat(itemToEat)
+   * -----------------------------
+   * Player eats a food item, restoring their health.
+   *
+   * Player can only eat Food instances.
+   * Player can only eat food items from their pack.
+   *
+   * Remove itemToEat from the pack.
+   * Increase the player's health by the food's energy amount, but do not
+   *   exceed the player's max health.  If exceeded, simply set player's health
+   *   to max health instead.
+   * To access the player's max health, be sure to use Player's getMaxHealth method.
+   * You should be able to invoke this function on a Player instance.
+   *
+   * @name eat
+   * @param {Food} itemToEat  The food item to eat.
+   */
+
+  this.eat = function(itemToEat) {
+    let isFood = itemToEat instanceof Food;
+    if (!isFood || this._pack.indexOf(itemToEat) === -1) {
+      return false;
+    }
+    this._pack.splice(this._pack.indexOf(itemToEat), 1);
+    if (this.health + itemToEat.energy > 100) {
+      this.health = 100;
+    } else {
+      this.health = this.health + itemToEat.energy;
+    }
+  };
+
+  /**
+   * Player Class Method => useItem(item)
+   * -----------------------------
+   * Player uses an item from the pack.
+   *
+   * If the item is a weapon, the player should equip the item.
+   * If the item is food, the player should eat the item.
+   * You should be able to invoke this function on a Player instance.
+   *
+   * @name useItem
+   * @param {Item/Weapon/Food} item   The item to use.
+   */
+
+  this.useItem = function(item) {
```

Comment:
This should be place outside the function and set on this classes prototype. Right now, you're not setting it on the prototype but on the object instance.